x-pack/solutions/security/plugins/security_solution/public/data_view_manager/redux/listeners/data_view_selected.test.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createDataViewSelectedListener } from './data_view_selected';
+import { selectDataViewAsync } from '../actions';
+import type { DataViewsServicePublic, FieldSpec } from '@kbn/data-views-plugin/public';
+import type { AnyAction, Dispatch, ListenerEffectAPI } from '@reduxjs/toolkit';
+import type { RootState } from '../reducer';
+import { DataViewManagerScopeName } from '../../constants';
+
+const mockDataViewsService = {
+  get: jest.fn(),
+  create: jest.fn().mockResolvedValue({
+    id: 'adhoc_test-*',
+    isPersisted: () => false,
+    toSpec: () => ({ id: 'adhoc_test-*', title: 'test-*' }),
+  }),
+} as unknown as DataViewsServicePublic;
+
+const mockedState: RootState = {
+  dataViewManager: {
+    analyzer: {
+      dataView: null,
+      status: 'pristine',
+    },
+    timeline: {
+      dataView: null,
+      status: 'pristine',
+    },
+    default: {
+      dataView: null,
+      status: 'pristine',
+    },
+    detections: {
+      dataView: null,
+      status: 'pristine',
+    },
+    shared: {
+      adhocDataViews: [
+        {
+          id: 'adhoc_test-*',
+          title: 'test-*',
+          fields: {
```

Comment:
So we will store the fields in the reducers?

Reply:
just temporarily, the goal is to remove it